Editor: Prevent wrapping text when showing icon labels in header

aaronrobertshaw opened the pull request:
Fixes: https://github.com/WordPress/gutenberg/issues/65491

## What?

Prevents wrapping text for editor header buttons when `showIconLabels` is set to `true`.

## Why?

Preventing wrapping keeps these buttons in line with others within the header and arguably more readable. 

See https://github.com/WordPress/gutenberg/issues/65491 for more details.

## How?

Adds `white-space: nowrap` for buttons within the editor's header while the show icon labels preference is enabled.

## Testing Instructions

1. Open editor and enable the show button text labels preference under Accessibility.
    <img width="350" alt="Screenshot 2024-10-11 at 3 59 44 pm" src="https://github.com/user-attachments/assets/69604281-fa7d-48df-8f5a-f1ff4eea26ba">
2. Confirm that the "Zoom out" button in the editor's header does not wrap at any viewport.
3. For bonus points, switch languages or add pinned plugin sidebars and confirm no wrapping occurs.

<details>
<summary>Test snippet to add pinned plugin</summary>

Paste the following into dev tools:

```js
(function (wp, React) {
    var el = React.createElement;
    var registerPlugin = wp.plugins.registerPlugin;
    var PluginSidebar = wp.editor.PluginSidebar;
    var __ = wp.i18n.__;

    const SimplePluginSidebar = () => {
        return el(
            PluginSidebar,
            {
                name: 'simple-plugin-sidebar',
                title: __('My Simple Sidebar'),
                icon: 'smiley',
            },
            el('p', null, __('This is a simple plugin sidebar.'))
        );
    };

    // Register the plugin
    registerPlugin('simple-plugin-sidebar', {
        render: SimplePluginSidebar,
    });
})(window.wp, window.React);
```

</details>

## Screenshots or screencast <!-- if applicable -->

Before: 
<img width="1289" alt="Screenshot 2024-10-11 at 3 38 14 pm" src="https://github.com/user-attachments/assets/df588e7a-7a03-4e9e-9def-764b81f10843">

After: 
<img width="1288" alt="Screenshot 2024-10-11 at 3 38 34 pm" src="https://github.com/user-attachments/assets/71e9021e-56d3-40b7-83ee-7deea8046ff0">

aaronrobertshaw committed:
Editor: Prevent wrapping text when showing icon labels in header

afercia commented:
On https://github.com/WordPress/gutenberg/issues/65491#issuecomment-2406713485 I pinged the @WordPress/gutenberg-components to ask why the `default` button does not use `white-space: nowrap;` in the first place. The primary, secondary, and tertiary variants do. Setting `white-space: nowrap;` on the default style would inherit to the CSS generated element used for `showIconLabels` and appears to be slighgly more solid. Unless there are specific reasons to not do that, I'd consider to apply this fix to the default style.

ciampo commented:
Opened https://github.com/WordPress/gutenberg/issues/66049 to discuss it

tyxla commented:
I'd be fine to ship this as a temporary fix while debating the best way forward for handling this more broadly in `<Button />`.

ciampo commented:
Makes sense. If we do add the style directly to `Button`, we could remove the temporary fix.

aaronrobertshaw commented:
Thanks for the discussion here.

> Setting white-space: nowrap; on the default style would inherit to the CSS generated element used for showIconLabels and appears to be slighgly more solid.

I'm not sure I agree 100%. If we were starting from a clean slate, it makes perfect sense. My concern is adding `nowrap` to default buttons after the fact might be a breaking change for some consumers. That's partly why I erred towards the smaller scope of this quick fix.

On that note, it appears that we have consensus to land this fix for 6.7 and follow-up more holistically via https://github.com/WordPress/gutenberg/issues/66049

Once this PR gets a formal approval, I'll get it merged.

andrewserong reviewed (approved):
> I'm not sure I agree 100%. If we were starting from a clean slate, it makes perfect sense. My concern is adding nowrap to default buttons after the fact might be a breaking change for some consumers. That's partly why I erred towards the smaller scope of this quick fix.

I like the idea of handling this at the leaf level, as it were, especially since this is a fix for 6.7. This is testing well for me, I notice that Zoom Out hasn't been translated yet in a lot of languages, so testing in Spanish didn't yield results different to English for this button, but overall this LGTM!

![image](https://github.com/user-attachments/assets/e6e83e0e-f353-4e87-b934-50a3eff29cfa)

ramonjd reviewed (approved):
LGTM

| Before | After |
|--------|--------|
| <img width="300" alt="Screenshot 2024-10-14 at 1 30 06 pm" src="https://github.com/user-attachments/assets/45ef1814-b71a-43d3-b9ea-442305f952e6"> | <img width="300" alt="Screenshot 2024-10-14 at 1 30 12 pm" src="https://github.com/user-attachments/assets/4a44f386-220b-4f68-8b97-b8895e389959"> | 


Checked some other locales:

German doesn't appear to be translated, but the extra room the other translations takes up looks okay
<img width="561" alt="Screenshot 2024-10-14 at 1 33 50 pm" src="https://github.com/user-attachments/assets/a7ddf0ca-ddb0-426e-88a6-f26875806e71">


Chinese characters and RTL don't wrap

aaronrobertshaw commented:
Given this was more of a visual bug and there is a desire to see it land in 6.7 as a bug fix. I've corrected the labels on this PR and added `Backport to WP 6.7 Beta/RC`

ciampo commented:
> I'm not sure I agree 100%. If we were starting from a clean slate, it makes perfect sense. My concern is adding `nowrap` to default buttons after the fact might be a breaking change for some consumers. That's partly why I erred towards the smaller scope of this quick fix.

It's a valid concern, and I think we're in a bit of a gray area. This change could also be interpreted as a bug fix, where the default variant was (mistakenly) left behind when setting the text wrapping styles in the component.

In any case, the change to the default variant is proposed in https://github.com/WordPress/gutenberg/issues/66049 and therefore I encourage everyone to help identify pros/cons of the approach (and unwanted side-effects) in that PR 🙏
